### PR TITLE
change style file extension from QML to QLP

### PR DESCRIPTION
**Pull request by alexbruy**
This fixes overlap with QtQuick/QtQML files and also reflects that this is more than just style, as now it contains labeling settings, widgets settings etc.

I only chaged corresponding strings and some variable names. Most of the variables and names of the fields in the databases left unchanged (e.g. when saving style to database). Not sure if we should change them too.

### CODE REVIEW

File: src/app/qgsrasterlayerproperties.cpp [hunk 1]
```diff
@@ -1762,14 +1762,10 @@ void QgsRasterLayerProperties::loadStyle_clicked()
                        this,
                        tr( "Load layer properties from style file" ),
                        lastUsedDir,
-                       tr( "QGIS Layer Style File" ) + " (*.qml)" );
+                       tr( "QGIS Layer Properties File" ) + " (*.qlp);;" + tr( "QGIS Layer Style File" ) + " (*.qml)" );
```
Comment (m-kuhn) [inline src/app/qgsrasterlayerproperties.cpp]: Should we rephrase this to "QGIS Layer Style File (deprecated)" to reduce the confusion about "what's the difference between the two?"?
Is the double `;;` on purpose?

Comment (alexbruy) [inline src/app/qgsrasterlayerproperties.cpp]: double `;;` is on purpose this is the syntax for filters in Qt file dialogs.

Comment (m-kuhn) [inline src/app/qgsrasterlayerproperties.cpp]: Or "QGIS 2.x Layer Style Files"?

**Comment by m-kuhn** [comment]
It would be nice to rename database stuff 2. I'm not involved enough to see if we can do automatic updating of this?

**Comment by alexbruy** [comment]
As I can see the only thing that needs updating in the database part is the name of the field `styleQML` and some internal variables. For example https://github.com/qgis/QGIS/blob/master/src/providers/postgres/qgspostgresprovider.cpp#L4403 or https://github.com/qgis/QGIS/blob/master/src/providers/postgres/qgspostgresprovider.cpp#L4448

**Comment by m-kuhn** [comment]
Do you think we should also ship an `ALTER TABLE layer_styles RENAME column` script? Detect if the user is still running a 2.x version and propose to update?

**Comment by jgrocha** [comment]
Hi,
If we update the `styleqml` column name, we might invalidate an environment where users are using QGIS 2.x and QGIS 3.x on top of the same database. If the table is created with QGIS 3.x, a QGIS 2.x user would have to delete the table manually, to recreate it for 2.x
At least, we should warn the users of this potential inconsistency.
We might end with much more noise if we change the column name, rather than keep it as it is.

**Comment by m-kuhn** [comment]
> we might invalidate an environment where users are using QGIS 2.x and QGIS 3.x on top of the same database

I'm not sure the contents (style XML) will be compatible in every case.
So I imagine we can even end up with better UX if we have two different columns for 2.x styles and 3.x style?

**Comment by jgrocha** [comment]
@m-kuhn Adding the `styleqlp` column if necessary is not difficult to implement and the user will not notice if the style comes from the `styleqml` or the `styleqlp` column. This can support an hybrid environment with QGIS 2.x and 3.x versions running on top of the same database.

**Comment by alexbruy** [comment]
As QGIS 3 will introduce a lot of API breaks, I think we can also break table schema. Moreover styles definition (and other parts of the QML/QLP file) may also change during QGIS 3.0 development.

**Comment by NathanW2** [comment]
I don't see breaking schema here as a good thing as QGIS 2 and 3 will share
the same DB connection.  Better to extend the table or make a new one just
for 3.0

**Comment by alexbruy** [comment]
Well, I can add another column to the `layer_styles` table when creating it from scratch and some logic to alter this table if it is already present in a database. But at some point we anyway should drop QML column from it otherwise we will end up with a huge table with columns for each QGIS version, because same DB may be used by QGIS 2, QGIS 3, QGIS 4....

**Comment by NathanW2** [comment]
I would say cross that bridge when it comes no need to do that now.  IMO
it's not a big enough deal to break the table.

If you break the table what is the solution if you want to use QGIS 2 and
QGIS 3 on the same DB?

**Comment by m-kuhn** [comment]
> I can add another column to the layer_styles table when creating it from scratch and some logic to alter this table if it is already present in a database.

That sounds good to me.

Apart from the database pollution, which only happens when migrating and not when creating from scratch, I think that sound like a sane approach?

**Comment by m-kuhn** [comment]
@alexbruy any news on this, it would be a nice change for 3.0?
